dex/Dockerfile

```
@@ -1,9 +1,9 @@
 # dex container image
 
 # Stage1: build from source
-FROM ghcr.io/cybozu/golang:1.21-jammy AS build
+FROM ghcr.io/cybozu/golang:1.22-jammy AS build
 
-ARG DEX_VERSION=2.37.0
+ARG DEX_VERSION=2.38.0
```

See the note in maintenance.md.

> This image is used by [ArgoCD](#argocd).
> So browse the following manifest and check the required version.
> If the manifest uses version _a.b.c_, we should use version _a.b.d_ where _d >= c_.
> Don't use a newer minor version.

The manifest below uses `2.37.0`.
So we can use one of `2.37.0`, `2.37.1`, `2.37.2`, ..., if exists.
We cannot use `2.38.0` because it may be incompatible.

https://github.com/argoproj/argo-cd/blob/v2.10.1/manifests/base/dex/argocd-dex-server-deployment.yaml#L40

```suggestion
ARG DEX_VERSION=2.37.0
```